lib/kamal/cli/build.rb

```diff
@@ -115,7 +115,7 @@ def connect_to_remote_host(remote_host)
       remote_uri = URI.parse(remote_host)
       if remote_uri.scheme == "ssh"
         options = { user: remote_uri.user, port: remote_uri.port }.compact
-        on(remote_uri.host, options) do
+        on(remote_host.gsub('ssh://', ''), options) do
```

Comment:
I think we could do:

```ruby
host = SSHKit::Host.new(
  hostname: remote_uri.host,
  ssh_options: { user: remote_uri.user, port: remote_uri.port }
)
on(host) do
  execute "true"
end
```

which would avoid needing to alter the options in sshkit_with_ext.rb

Reply:
Good catch, I've updated PR accordingly